File: 2392-successful-pairs-of-spells-and-potions/2392-successful-pairs-of-spells-and-potions.cpp
class Solution {
public:
    vector<int> successfulPairs(vector<int>& spells, vector<int>& potions, long long success) {
        vector<int> result;
        int n = spells.size();
        int m = potions.size();
        sort(potions.begin(),potions.end());
        
        for(int i=0;i<n;i++)
        {
            int index=m;
            int left = 0;
            int right = m-1;
            while(left<=right)
            {
                int mid = (left+right)/2;
                if(((long long)spells[i]*potions[mid]) >= success)
                {
                index = mid;
                right=mid-1;
                }
                else
                {
                left=mid+1;
                }
            }
            result.push_back(m-index);
        }
        return result;
    }
};

File: 2392-successful-pairs-of-spells-and-potions/2392-successful-pairs-of-spells-and-potions.py
class Solution:
    def successfulPairs(self, spells: List[int], potions: List[int], success: int) -> List[int]:
        vector = []
        n = len(spells)
        m = len(potions)
        potions.sort()
        for i in range(0,n,1):
            start = 0
            end = m-1
            index = m
            while start<=end:
                mid = (start+end)//2
                if spells[i]*potions[mid]>=success:
                    index = mid
                    end = mid-1
                else:
                    start = mid+1
            vector.append(m-index)    
        return vector